app/services/katello/repository_type.rb

```diff
@@ -154,11 +160,19 @@ def initialize(options)
         self.content_type = options[:content_type]
         self.filename_key = options[:filename_key]
         self.duplicates_allowed = options[:duplicates_allowed]
+        self.ui_config = options[:ui_config]
```

Comment:
Can we move this to parent ContentType so it's available for non-generic types too..You could add ui_config to to_json of parent and remove the overriding that limits it to generic types..Also would super() be helpful here for the common bits of initialize..